src/store/Account.ts

```diff
@@ -511,39 +513,30 @@ export default {
         { root: true },
       )
 
-      try {
-        // - prepare REST parameters
-        const repositoryFactory = rootGetters['network/repositoryFactory'] as RepositoryFactory
-        const transactionHttp = repositoryFactory.createTransactionRepository()
-
-        // - prepare scoped *confirmation listener*
-        const listener = repositoryFactory.createListener()
-        await listener.open()
-
-        // - announce hash lock transaction and await confirmation
-        transactionHttp.announce(signedLock)
-
-        // - listen for hash lock confirmation
-        return new Promise((resolve, reject) => {
-          const address = Address.createFromRawAddress(issuer)
-          return listener.confirmed(address).subscribe(
-            async () => {
-              // - hash lock confirmed, now announce partial
-              await transactionHttp.announceAggregateBonded(signedPartial)
-              commit('removeSignedTransaction', signedLock)
-              commit('removeSignedTransaction', signedPartial)
-              return resolve(new BroadcastResult(signedPartial, true))
-            },
-            () => {
-              commit('removeSignedTransaction', signedLock)
-              commit('removeSignedTransaction', signedPartial)
-              reject(new BroadcastResult(signedPartial, false))
-            },
-          )
-        })
-      } catch (e) {
-        return new BroadcastResult(signedPartial, false, e.toString())
-      }
+      // - prepare REST parameters
+      const repositoryFactory = rootGetters['network/repositoryFactory'] as RepositoryFactory
+      const transactionHttp = repositoryFactory.createTransactionRepository()
+      const receiptsHttp = repositoryFactory.createReceiptRepository()
+      const transactionService = new TransactionService(transactionHttp, receiptsHttp)
+      // - prepare scoped *confirmation listener*
+      const listener = repositoryFactory.createListener()
+      await listener.open()
```

Comment:
Why not reuse the network's listener? 
If you create a new listener, I won't mix promise and observable. Try creating observable from the wait promise and pipe it with the other observables.

BTW, I'm moving the transaction logic out of the account. I definitely add the new timeout in my new PR.